packages/design-system-react-native/src/components/Checkbox/Checkbox.types.ts

```diff
@@ -0,0 +1,75 @@
+import type { StyleProp, PressableProps, ViewProps } from 'react-native';
+
+import type { IconProps } from '../Icon';
+import type { TextProps } from '../Text';
+/**
+ * Checkbox component props.
+ */
+
+export type CheckboxProps = {
+  /**
+   * Optional prop that when true, shows a checked checkbox.
+   *
+   * @default false
+   */
+  isSelected?: boolean;
+
+  /**
+   * Optional prop for the initial state of checkbox that when true, shows a checked checkbox.
+   * Use this prop if you want Checkbox to automatically control the isSelected/checked state
+   *
+   * @default false
+   */
+  defaultIsSelected?: boolean;
+
+  /**
+   * Optional prop that when true, disables the checkbox
+   *
+   * @default false
+   */
+  isDisabled?: boolean;
+
+  /**
+   * Optional prop that when true, shows the invalid state
+   *
+   * @default false
+   */
+  isInvalid?: boolean;
+
+  /**
+   * Optional label prop
+   */
+  label?: React.ReactNode | string;
+
+  /**
+   * Optional props to be passed to the label's Text component
+   */
+  labelProps?: Omit<Partial<TextProps>, 'children'>;
+
+  /**
+   * Optional callback for when the checked state changes
+   *
+   * @default false
+   */
+  onChange?: (isSelected: boolean) => void;
+
+  /**
+   * Optional props to be passed to the checkbox container component
+   */
+  checkboxContainerProps?: Omit<Partial<ViewProps>, 'children'>;
+
+  /**
+   * Optional props to be passed to the label's Text component
+   */
+  checkedIconProps?: Partial<IconProps>;
+
+  /**
+   * Optional prop to add twrnc overriding classNames.
+   */
+  twClassName?: string;
+
+  /**
+   * Optional prop to control the style.
+   */
+  style?: StyleProp<PressableProps>;
```

Comment:
`ViewStyle` here should fix the type error in the tests and I think aligns with other react native components

```suggestion
  style?: StyleProp<ViewStyle>;
```

Reply:
nope this has to be `PressableProps` style because the wrapping container is `Pressable`, not `View`

Reply:
the other place needs to be updated

Reply:
I noticed we get a type error when using the `style` prop though?

<img width="637" alt="Screenshot 2025-05-22 at 12 42 02 PM" src="https://github.com/user-attachments/assets/eb844098-7a54-4abc-9a24-36c972f88d5b" />

<img width="433" alt="Screenshot 2025-05-22 at 12 42 08 PM" src="https://github.com/user-attachments/assets/64681cbe-35e3-4004-ad67-797ea21fb62a" />